### PR TITLE
 Fix "Protected Terms" formatting #10415

### PR DESCRIPTION
**Pull request by yuyan-z**
Fixes https://github.com/JabRef/jabref/issues/10415. 
1. Remove unwanted leading and trailing whitespaces while adding the selected text to list, as required in Step 6.
2. Remove unwanted empty brackets while formatting the text, as required in Step 14.
3. Allow the user to add the word at the cursor, as required in Step 4. For example, 
image
image
image

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
I tried it. Looks good.

One more thing:

1. Double click "BPMN4TOSCA"
2. Select "Protect terms" -> "Protect selection"\
    image

Here, the space should also not be protected

image

Could you include a fix for that, too?

**Comment by yuyan-z** [comment]
> I tried it. Looks good.
> 
> One more thing:
> 
> 1. Double click "BPMN4TOSCA"
> 2. Select "Protect terms" -> "Protect selection"
>    image
> 
> Here, the space should also not be protected
> 
> image
> 
> Could you include a fix for that, too?

Thank you for the comments! I updated the code to fix this issue. 
Now it can process the selected text containing leading or trailing spaces. For example:
image
image
image

File: CHANGELOG.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We fixed an issue where there was a failure to access the url link for "eprint" for the ArXiv entry.[#10474](https://github.com/JabRef/jabref/issues/10474)
 - We fixed an issue where it was not possible to connect to a shared database once a group with entries was added or other metadata modified [#10336](https://github.com/JabRef/jabref/issues/10336)
 - We fixed an issue where middle-button paste in X not always worked [#7905](https://github.com/JabRef/jabref/issues/7905)
+- We fixed an issue where the added protected term has unwanted leading and trailing whitespaces, where the formatted text has unwanted empty brackets and where the word after or at the cursor in the textbox can be added to the list [#10415](https://github.com/JabRef/jabref/issues/10415).
```
Comment (Siedlerchr) [inline CHANGELOG.md:71]: As we released 5.11 in the meantine, you need to move this to the unreleased secton

**Review by koppor — APPROVED** [review]
For sure, test cases would be good. Especially at these low-level string operations. You can do these as follow-up if you want.

For now, good to go!

**Pull request merged**